### PR TITLE
MINIFICPP-2326 Link lua statically

### PR DESCRIPTION
**Pull request by fgerlits**
Link liblua statically into the lua script extension instead of using the lua library installed on the system.

https://issues.apache.org/jira/browse/MINIFICPP-2326

---

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [x] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] If applicable, have you updated the LICENSE file?
- [x] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by lordgamez** [comment]
I think some other files need to be updated as well:
- Remove Lua from build time dependencies in README.md
- Remove lua package installation from bootstrap shell files if lua extension is enabled (aptitude.sh, arch.sh, centos.sh, darwin.sh, debian.sh, fedora.sh, rheldistro.sh, suse.sh)
- Remove lua package installation from python bootstrap if lua extension is enabled (package_manager.py)
- Remove lua development packages from dockerfiles

**Comment by fgerlits** [comment]
> I think some other files need to be updated as well

Yes, good point, thanks!  Fixed in 450f397604263c1faafdfffdd3aab7158556720e.

File: libminifi/test/unit/CronTests.cpp [hunk 1]
```diff
@@ -421,7 +421,7 @@ TEST_CASE("Cron::calculateNextTrigger with timezones", "[cron]") {
   date::set_install(TZ_DATA_DIR);
 #endif
 
-  const std::vector<std::string> time_zones{ "Europe/Berlin", "Asia/Seoul", "America/Los_Angeles", "Asia/Singapore", "UCT" };
+  const std::vector<std::string> time_zones{ "Europe/Berlin", "Asia/Seoul", "America/Los_Angeles", "Asia/Singapore", "UTC" };
```
[the remark this answers is not in the document]
Comment (fgerlits) [reply libminifi/test/unit/CronTests.cpp:424]: Continuing to support UCT would be a bigger piece of work, if we want to do it.

Starting with Ubuntu mantic and Debian trixie (and probably other newer distros), UCT is no longer a top level time zone in `/usr/share/zoneinfo`, but an alias to UTC under `/usr/share/zoneinfo/Etc`.  The `date` library does not like aliases: in [tzdb::locate_zone](https://github.com/HowardHinnant/date/blob/cd3c57932f7f3906c5876b8f04e2ff6b20d2070b/src/tz.cpp#L3730), it finds that `tz_name` (= "UCT") is different from the name of the time zone it points to (= "UTC") and it fails.

[the remark this answers is not in the document]
Comment (szaszm) [reply libminifi/test/unit/CronTests.cpp:424]: If it was just a typo, I'm OK with dropping it. My distro is quite 'new', an Arch Linux updated last week, and it still has "UCT" under `/usr/share/zoneinfo` (and also under `Etc`), which is not a symlink, but identical to "UTC". So I'm guessing it's a debian/ubuntu thing. Either way, the test should work on those systems.

Updating the date library or moving to the C++20 would be nice, but I'd keep it separate from this PR. If you're investigating, please share your results, but I don't want to hold up this PR with the additional change, if it's not necessary.

[the remark this answers is not in the document]
Comment (martinzink) [reply libminifi/test/unit/CronTests.cpp:424]: I didnt have a strong opinion on this when I wrote it to ne honest didnt know about this whole situation before this thread and now that I looked it up. Im good with changing it 👍

**Review by martinzink — APPROVED** [review]

File: libminifi/test/unit/CronTests.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (szaszm) [reply libminifi/test/unit/CronTests.cpp:424]: ok, then let's leave it changed :)

**Review by szaszm — APPROVED** [review]

**Review by lordgamez — APPROVED** [review]

**Pull request closed without merging**